### PR TITLE
Jsm solution

### PR DESCRIPTION
**Pull request by jsm13**
completeness: 4
comfort: 4

questions: Was trying to use an updateDisplay() helper function which despite logging the exact string I wanted, would update the displays to '$undefined'. Good chance looking at this with rested eyes tomorrow the reason will slap me in the face but at the moment I'm pretty stumped.

needing fixing: 
- .zero class not applied to initial balance of zero
- overdraft protection
- code is soaking wet

### CODE REVIEW

File: atm.js [hunk 1]
```diff
@@ -1,21 +1,78 @@
-var checkingBalance = document.querySelector("div#checking_balance");
-checkingBalance = parseInt( checkingBalance.innerHTML.replace("$", "") );
-
-var checkingDeposit = function(amount){
-  checkingBalance = parseInt( checkingBalance.innerHTML.replace("$", "") );
-
+var atm = {
```
Comment (jshawl) [inline atm.js:1]: Excellent use of object here!!

File: atm.js [hunk 2]
```diff
@@ -1,21 +1,78 @@
...
+      var account = $(event.target).prevAll("h2").text();
+      var amount = $(event.target).prevAll("[type=text]").val();
+      $(event.target).prevAll("[type=text]").val("");
+      if (action == "Withdraw"){this.withdraw(account, amount)}
+      else if (action == "Deposit"){this.deposit(account, amount)}
+    }.bind(this));
+  },
+  withdraw: function(account, amount){
+    if (parseFloat(amount) > this.balance[account.toLowerCase()]){
+      alert("overdraft protection coming soon!");
```
Comment (jshawl) [inline atm.js:18]: this is great user experience! :)

File: atm.js [hunk 3]
```diff
@@ -1,21 +1,78 @@
...
+      return;
+    }
+    else if (isNaN(parseFloat(amount))){
```
Comment (jshawl) [inline atm.js:21]: excellent job handling bad user input!

File: atm.js [hunk 4]
```diff
@@ -1,21 +1,78 @@
...
+    console.log(this.balance[account.toLowerCase()]);
+    $("#"+account.toLowerCase()+ " .balance")[0].textContent = ("$" + this.balance[account.toLowerCase()]);
+    if (this.balance[account.toLowerCase()] > 0){$("#"+account.toLowerCase()+ " .balance").removeClass("zero");}
+  }
+  // updateDisplay: function(){
+  //   console.log($("#checking .balance"));
+  //   var cb = this.balance.checking;
+  //   console.log("$" + this.balance.checking);
+  //   $("#checking .balance")[0].textContent = ("fucking anything!");
+  //   $("#saving, .balance").html("$" + this.balance.saving)
```
Comment (jshawl) [inline atm.js:44]: maybe this comma is preventing your code from working as intended?

File: atm.js [hunk 5]
```diff
@@ -1,21 +1,78 @@
...
-function savingsWithdrawal(amount){
-
-}
-
-// an eventListerner for each button, each one a "click"
\ No newline at end of file
+atm.listen();
+//
+//
+// var checking = {
```
Comment (jshawl) [inline atm.js:50]: please remove commented out code before submitting in the future. If you want to keep it for reference, you can use a git branch to store.

**Comment by jshawl** [comment]
:+1: :star2: Excellent job on this assignment!!! This solution is one of the dryest i've seen so far! re: your issues with the `updateDisplay` method - I'm not seeing any obvious problems. Let me know if you'd like to work through this in our one-on-one

**Comment by jsm13** [comment]
Hi Jesse, thanks so much! Really appreciate the feedback. I would be really
interested in talking about it in our one-by-one. Hope you're enjoying your
weekend. I'll be sure to keep submissions free of commented out code.

Best,

John

**Pull request closed without merging**